src/Model/Gallery.php

```diff
@@ -152,15 +152,13 @@ public function removeGalleryItem(GalleryItemInterface $galleryItem): void
      */
     public function reorderGalleryItems()
     {
-        if ($this->getGalleryItems() && $this->getGalleryItems() instanceof \IteratorAggregate) {
-            $iterator = $this->getGalleryItems()->getIterator();
-            \assert($iterator instanceof \ArrayIterator);
+        $iterator = $this->getGalleryItems()->getIterator();
+        \assert($iterator instanceof \ArrayIterator);
```

Comment:
Why is this assert needed ?

Reply:
Otherwise the uasort method is not detected:

```
 ------ -----------------------------------------------------------------------------------------------------------------
  Line   src/Model/Gallery.php
 ------ -----------------------------------------------------------------------------------------------------------------
  157    Call to an undefined method Traversable<(int|string), Sonata\MediaBundle\Model\GalleryItemInterface>::uasort().
 ------ -----------------------------------------------------------------------------------------------------------------
```

Reply:
We cannot be sure that the iterator is an ArrayIterator. It's not mandatory by the Collection interface.
If someone use a custom implementation it won't work.

It would be better to use another implementation or at least to throw an exception if it's not an array iterator instead of just asserting it.

Reply:
Added an error throw. Anyway I don't think adding own collection implementation is something that can be done.